r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -725,6 +725,29 @@ test_that("errors in strptime", {
   )
 })
 
+test_that("arrow_find_substring", {
+  
+  df <- tibble(x = c("Foo and Bar", "baz and qux and quux"))
+  
+  expect_equivalent(
+    df %>%
+      Table$create() %>%
+      mutate(x = arrow_find_substring(x, options = list(pattern = "b"))) %>%
+      collect(),
+    tibble(x = c(-1, 0))
+  )
+  
+  skip("ARROW-13157: find_substring not implemented with ignore_case")
+  expect_error(
+    df %>%
+      Table$create() %>%
+      mutate(x = arrow_find_substring(x, options = list(pattern = "b", ignore_case = TRUE))) %>%
+      collect(),
+    "NotImplemented: find_substring with ignore_case"
+  )
```

Comment:
Let's change this to a positive expectation so that after ARROW-13157 is resolved we can simply remove the `skip()` and it should succeed.
```suggestion
  expect_equivalent(
    df %>%
      Table$create() %>%
      mutate(x = arrow_find_substring(x, options = list(pattern = "b", ignore_case = TRUE))) %>%
      collect(),
    tibble(x = c(8, 0))
  )
```